commit c2e1ed54e1a3f613bdda3f085fee5b21c9fa174d
Author: Hein-Pieter van Braam-Stewart <hp@tmm.cx>
Date:   Sat Dec 14 23:16:19 2019 +0100

    Workaround

diff --git a/mcs/build/platforms/win32.make b/mcs/build/platforms/win32.make
index dd969b51..bb15aba2 100644
--- a/mcs/build/platforms/win32.make
+++ b/mcs/build/platforms/win32.make
@@ -9,10 +9,11 @@ PLATFORM_TEST_HARNESS_EXCLUDES = NotOnWindows
 
 EXTERNAL_RUNTIME = mono
 
-PLATFORM_CHANGE_SEPARATOR_CMD=tr '/' '\\\\'
+#PLATFORM_CHANGE_SEPARATOR_CMD=tr '\\' '/'
+PLATFORM_CHANGE_SEPARATOR_CMD=cat
 PLATFORM_PATH_SEPARATOR = ;
 
-override CURDIR:=$(shell cygpath -m $(CURDIR))
+#override CURDIR:=$(shell cygpath -m $(CURDIR))
 
 hidden_prefix = 
 hidden_suffix = .tmp
diff --git a/mcs/build/profiles/build.make b/mcs/build/profiles/build.make
index b1c9a410..004f9f33 100644
--- a/mcs/build/profiles/build.make
+++ b/mcs/build/profiles/build.make
@@ -32,7 +32,7 @@ endif
 #
 ILASM = $(PROFILE_RUNTIME) $(RUNTIME_FLAGS) $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/tmp/ilasm.exe
 STRING_REPLACER = $(PROFILE_RUNTIME) $(RUNTIME_FLAGS) $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/tmp/cil-stringreplacer.exe
-GENSOURCES =$(PROFILE_RUNTIME) $(RUNTIME_FLAGS) $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/tmp/gensources.exe
+GENSOURCES = mono $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/tmp/gensources.exe
 
 MCS = $(BOOTSTRAP_MCS)
 
diff --git a/mcs/build/rules.make b/mcs/build/rules.make
index e57a636e..16f6cdc7 100644
--- a/mcs/build/rules.make
+++ b/mcs/build/rules.make
@@ -60,7 +60,7 @@ INTERNAL_CSC = CSC_SDK_PATH_DISABLED= $(RUNTIME) $(RUNTIME_FLAGS) $(CSC_RUNTIME_
 RESGEN = MONO_PATH="$(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH" $(RUNTIME) $(RUNTIME_FLAGS) $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/resgen.exe
 STRING_REPLACER = MONO_PATH="$(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH" $(RUNTIME) $(RUNTIME_FLAGS) $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/cil-stringreplacer.exe
 ILASM = MONO_PATH="$(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH" $(RUNTIME) $(RUNTIME_FLAGS) $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/ilasm.exe
-GENSOURCES = MONO_PATH="$(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH" $(RUNTIME) $(RUNTIME_FLAGS) $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/gensources.exe
+GENSOURCES = MONO_PATH="$(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH" mono $(topdir)/class/lib/$(BUILD_TOOLS_PROFILE)/gensources.exe
 
 depsdir = $(topdir)/build/deps
 
